UnityProject/Assets/Scripts/Furniture/TableBuilding.cs

```diff
@@ -0,0 +1,115 @@
+		return true;
+	}
+	public void ServerPerformInteraction(HandApply interaction)
+	{
+		if (interaction.TargetObject != gameObject) return;
+		if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Wrench))
+		{
+			ToolUtils.ServerUseToolWithActionMessages(interaction, 0.5f,
+						"You start deconstructing a table frame...",
+						$"{interaction.Performer.ExpensiveName()} starts deconstructing a table frame...",
+						"You are deconstructing a table frame.",
+						$"{interaction.Performer.ExpensiveName()} deconstructs a table frame.",
+						() => Disassemble(interaction));
+			SoundManager.PlayNetworkedAtPos("Wrench", gameObject.WorldPosServer(), 1f);
+			
+			return;
+		}
+		else if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.MetalSheet))
+		{
+			ToolUtils.ServerUseToolWithActionMessages(interaction, 0.5f,
+						"You start constructing a metal table...",
+						$"{interaction.Performer.ExpensiveName()} starts constructing a metal table...",
+						"You are constructing a metal table.",
+						$"{interaction.Performer.ExpensiveName()} constructs a metal table.",
+						() => SpawnTable(interaction, metalTable));
+			;
+			return;
+		}
+		else if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.GlassSheet))
+		{
+			ToolUtils.ServerUseToolWithActionMessages(interaction, 0.5f,
+						"You start constructing a glass table...",
+						$"{interaction.Performer.ExpensiveName()} starts constructing a glass table...",
+						"You are constructing a glass table.",
+						$"{interaction.Performer.ExpensiveName()} constructs a glass table.",
+						() => SpawnTable(interaction, glassTable));
+			SoundManager.PlayNetworkedAtPos("GlassHit", gameObject.WorldPosServer(), 1f);
+			
+			return;
+		}
+		else if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.WoodenPlank))
+		{
+			ToolUtils.ServerUseToolWithActionMessages(interaction, 0.5f,
+						"You start constructing a wooden table...",
+						$"{interaction.Performer.ExpensiveName()} starts constructing a wooden table...",
+						"You are constructing a wooden table.",
+						$"{interaction.Performer.ExpensiveName()} constructs a wooden table.",
+						() => SpawnTable(interaction, woodTable));
+			SoundManager.PlayNetworkedAtPos("wood3", gameObject.WorldPosServer(), 1f);
+			
+			return;
+		}
+
+	}
+	[Server]
+	private void Disassemble(HandApply interaction)
+	{
+		Spawn.ServerPrefab("Rods", gameObject.WorldPosServer(), count: 2);
+		Despawn.ServerSingle(gameObject);
+	}
+	[Server]
+	private void SpawnTable(HandApply interaction, LayerTile tableToSpawn)
+	{
+		var interactableTiles = InteractableTiles.GetAt(interaction.TargetObject.TileWorldPosition(), true);
+		Vector3Int cellPos = interactableTiles.WorldToCell(interaction.TargetObject.TileWorldPosition());
+		interaction.HandObject.GetComponent<Stackable>().ServerConsume(2);
```

Comment:
I suspect this will give an error if you have just one sheet, but will still build a table.
I reckon you should check if stackable.Amount is sufficient before doing any build logic

Reply:
`public void ServerConsume(int consumed)
	{
		if (consumed > amount)
		{
			Logger.LogErrorFormat("Consumed amount {0} is greater than amount in this stack {1}, will not consume.",
				 Category.Inventory, consumed, amount);
			return;
		}
		SyncAmount(amount, amount - consumed);
		if (amount <= 0)
		{
			Despawn.ServerSingle(gameObject);
		}
	}`

Reply:
Is another check neccesary to put?